pkg/ottl/paths.go

```diff
@@ -30,3 +34,53 @@ func getBooleanExpressionPaths(be *booleanExpression) []path {
 	be.accept(visitor)
 	return visitor.paths
 }
+
+// AppendStatementPathsContext changes the given OTTL statement adding the context name prefix
+// to all context-less paths. No modifications are performed for paths which [Path.Context]
+// value matches any WithPathContextNames value.
+// The context argument must be valid WithPathContextNames value, otherwise an error is returned.
+func (p *Parser[K]) AppendStatementPathsContext(context string, statement string) (string, error) {
```

Comment:
Also a nit, but can we move this into the parser.go file? It is nice to keep all a struct's methods together if we can.